executor/join_test.go

```diff
@@ -944,47 +944,17 @@ func (s *testSuite2) TestHashJoin(c *C) {
 	tk.MustQuery("select count(*) from t2").Check(testkit.Rows("0"))
 	tk.MustExec("set @@tidb_max_chunk_size=1;")
 	result := tk.MustQuery("explain analyze select /*+ TIDB_HJ(t1, t2) */ * from t1 where exists (select a from t2 where t1.a = t2.a);")
-	// id	count	task	operator info	execution info
-	// HashLeftJoin_9	8000.00	root	semi join, inner:TableReader_13, equal:[eq(test.t1.a, test.t2.a)]	time:1.036712ms, loops:1, rows:0
-	// ├─TableReader_11	10000.00	root	data:TableScan_10	time:441.096µs, loops:1, rows:1
-	// │ └─TableScan_10	10000.00	cop	table:t1, range:[-inf,+inf], keep order:false, stats:pseudo
-	// └─TableReader_13	10000.00	root	data:TableScan_12	time:212.376µs, loops:1, rows:0
-	//   └─TableScan_12	10000.00	cop	table:t2, range:[-inf,+inf], keep order:false, stats:pseudo
+	// HashLeftJoin_9 7992.00 root semi join, inner:TableReader_15, equal:[eq(test.t1.a, test.t2.a)] time:219.863µs, loops:1, rows:0
+	// ├─TableReader_12 9990.00 root data:Selection_11 time:9.129µs, loops:1, rows:1
+	// │ └─Selection_11 9990.00 cop not(isnull(test.t1.a))
+	// │   └─TableScan_10 10000.00 cop table:t1, range:[-inf,+inf], keep order:false, stats:pseudo time:0s, loops:0, rows:5
+	// └─TableReader_15 9990.00 root data:Selection_14 time:12.983µs, loops:1, rows:0
+	//   └─Selection_14 9990.00 cop not(isnull(test.t2.a))
+	//       └─TableScan_13 10000.00 cop table:t2, range:[-inf,+inf], keep order:false, stats:pseudo time:0s, loops:0, rows:0
 	row := result.Rows()
-	c.Assert(len(row), Equals, 5)
+	c.Assert(len(row), Equals, 7)
 	outerExecInfo := row[1][4].(string)
 	c.Assert(outerExecInfo[len(outerExecInfo)-1:], Equals, "1")
-	innerExecInfo := row[3][4].(string)
+	innerExecInfo := row[4][4].(string)
 	c.Assert(innerExecInfo[len(innerExecInfo)-1:], Equals, "0")
-
-	tk.MustExec("insert into t2 select * from t1;")
-	tk.MustExec("delete from t1;")
```

Comment:
Because these 2 tests seem meaningless, `t2` contains 5 rows, `t1` is empty now, the assert conditions of line 974 and line 976 would always be true; same applies to the second test, and it is `explain` instead of `explain analyze` ...